plugins/modules/storagebox_set_password.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025 Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: storagebox_set_password
+short_description: (Re)set the password for a storage box
+version_added: 2.1.0
+author:
+    - Matthias Hurdebise (@matthiashurdebise)
+description:
+    - (Re)set the password for a storage box.
+extends_documentation_fragment:
+    - community.hrobot.robot
+    - community.hrobot.attributes
+    - community.hrobot.attributes.actiongroup_robot
+
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+  idempotent:
+    support: none
+    details:
+      - This module performs an action on every invocation.
+
+options:
+    id:
+        description:
+        - The ID of the storage box to modify.
+        type: int
+        required: true
+    password:
+        description:
+        - The new password for the storage box.
+        type: str
+        required: false
+"""
+
+EXAMPLES = r"""
+- name: Set the password
+  community.hrobot.storagebox_set_password:
+      id: 123
+      password: "newpassword"
+
+- name: Set a random password
+  community.hrobot.storagebox_set_password:
+      id: 123
```

Comment:
You need to register `result` here for the next task to work:
```suggestion
    id: 123
  register: result
```